crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -307,8 +307,17 @@ impl CheckpointStore {
     }
 
     fn prune_local_summaries(&self) -> SuiResult {
-        self.locally_computed_checkpoints.clear()?;
-        info!("Pruned all local checkpoint summaries at end of epoch");
+        if let Some((last_local_summary, _)) = self
+            .locally_computed_checkpoints
+            .unbounded_iter()
+            .skip_to_last()
+            .next()
+        {
+            let mut batch = self.locally_computed_checkpoints.batch();
+            batch.delete_range(&self.locally_computed_checkpoints, &0, &last_local_summary)?;
```

Comment:
This makes sense, we also have `ignore_range_delete` set to true by default, so we shouldn't see any performance  impact on the read side from it.